plugins/destination/firehose/client/write.go

```diff
@@ -0,0 +1,133 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/arn"
+	"github.com/aws/aws-sdk-go-v2/service/firehose"
+	"github.com/aws/aws-sdk-go-v2/service/firehose/types"
+	"github.com/cloudquery/plugin-sdk/plugins/destination"
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+const MAX_RECORD_SIZE_BYTES = 1024000
+const MAX_BATCH_RECORDS = 500
+const MAX_BATCH_SIZE_BYTES = 4194000
```

Comment:
```suggestion
const MaxRecordSizeBytes = 1024000
const MaxBatchRecords = 500
const MaxBatchSizeBytes = 4194000
```

(nit to make it match general Go style. The names will also have to be updated lower down)